lib/rules/no-adjacent-inline-elements.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Prevent adjacent inline elements not separated by whitespace.
+ * @author Sean Hayes
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+// https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements
+var inlineNames = ('a b big i small tt abbr acronym cite code dfn em kbd strong' +
+  ' samp time var bdo br img map object q script span sub sup button input' +
+  ' label select textarea').split(' ');
+var nbsp = '&nbsp;';
+
+function isInline(node) {
+  if (node.type === 'Literal') {
+    // Regular whitespace will be removed.
+    var raw = node.raw.trim();
+    // To properly separate inline elements, each end of the literal will need
+    // a &nbsp;.
+    return !(raw.startsWith(nbsp) && raw.endsWith(nbsp));
```

Comment:
also node 0.10 doesn't have startsWith or endsWith, so we'll want to use `.slice(0, 1)` and `.slice(-1)` here